### PR TITLE
Replaces the current profiler logger with spdlog.

### PR DESCRIPTION
**Pull request by tonyredondo**
The PR removes the current profiler logger implementation, with stdlog lib.
#### Note: This PR also fixes the issue where the profiler log reports x86 even when the process is x64

The format for logs is:
`{time} [{process name}|{pid}|{threadId}] [{level}] {message}`

#### Example from CI:
```
10/30/20 02:43:14.415 PM [dotnet|306|306] [info] Datadog CLR Profiler 1.19.6 64-bit
10/30/20 02:43:14.415 PM [dotnet|305|305] [info] Environment variables:
10/30/20 02:43:14.415 PM [dotnet|306|306] [info] Environment variables:
10/30/20 02:43:14.415 PM [dotnet|305|305] [info]   DD_TRACE_ENABLED=
10/30/20 02:43:14.415 PM [dotnet|306|306] [info]   DD_TRACE_ENABLED=
10/30/20 02:43:14.415 PM [dotnet|305|305] [info]   DD_TRACE_DEBUG=
10/30/20 02:43:14.415 PM [dotnet|306|306] [info]   DD_TRACE_DEBUG=
10/30/20 02:43:14.415 PM [dotnet|305|305] [info]   DD_DUMP_ILREWRITE_ENABLED=
10/30/20 02:43:14.415 PM [dotnet|306|306] [info]   DD_DUMP_ILREWRITE_ENABLED=
10/30/20 02:43:14.415 PM [dotnet|306|306] [info]   DD_DOTNET_TRACER_HOME=/project/src/Datadog.Trace.ClrProfiler.Native/bin/Debug/x64
10/30/20 02:43:14.415 PM [dotnet|305|305] [info]   DD_DOTNET_TRACER_HOME=/project/src/Datadog.Trace.ClrProfiler.Native/bin/Debug/x64
10/30/20 02:43:14.415 PM [dotnet|306|306] [info]   DD_INTEGRATIONS=/project/test/test-applications/integrations/Samples.StackExchange.Redis/bin/Debug/netcoreapp2.1/publish/profiler-lib/integrations.json
10/30/20 02:43:14.415 PM [dotnet|306|306] [info]   DD_PROFILER_PROCESSES=dotnet
10/30/20 02:43:14.415 PM [dotnet|306|306] [info]   DD_PROFILER_EXCLUDE_PROCESSES=
10/30/20 02:43:14.415 PM [dotnet|306|306] [info]   DD_AGENT_HOST=
10/30/20 02:43:14.415 PM [dotnet|305|305] [info]   DD_INTEGRATIONS=/project/test/test-applications/integrations/Samples.MongoDB/bin/Debug/netcoreapp2.1/publish/profiler-lib/integrations.json
10/30/20 02:43:14.415 PM [dotnet|305|305] [info]   DD_TRACE_AGENT_PORT=35511
10/30/20 02:43:14.415 PM [dotnet|306|306] [info]   DD_CLR_DISABLE_OPTIMIZATIONS=
10/30/20 02:43:14.415 PM [dotnet|306|306] [info]   DD_TRACE_DOMAIN_NEUTRAL_INSTRUMENTATION=
10/30/20 02:43:14.415 PM [dotnet|306|306] [info]   DD_TRACE_NETSTANDARD_ENABLED=
10/30/20 02:43:14.415 PM [dotnet|306|306] [info]   DD_AZURE_APP_SERVICES=
10/30/20 02:43:14.415 PM [dotnet|305|305] [info]   DD_ENV=
10/30/20 02:43:14.415 PM [dotnet|306|306] [info]   APP_POOL_ID=
10/30/20 02:43:14.415 PM [dotnet|306|306] [info]   DOTNET_CLI_TELEMETRY_PROFILE=
10/30/20 02:43:14.415 PM [dotnet|305|305] [info]   DD_SERVICE=
10/30/20 02:43:14.415 PM [dotnet|305|305] [info]   DD_VERSION=1.0.0
10/30/20 02:43:14.415 PM [dotnet|305|305] [info]   DD_DISABLED_INTEGRATIONS=
10/30/20 02:43:14.415 PM [dotnet|305|305] [info]   DD_CLR_DISABLE_OPTIMIZATIONS=
10/30/20 02:43:14.415 PM [dotnet|305|305] [info]   DD_TRACE_DOMAIN_NEUTRAL_INSTRUMENTATION=
10/30/20 02:43:14.415 PM [dotnet|305|305] [info]   DD_TRACE_NETSTANDARD_ENABLED=
10/30/20 02:43:14.415 PM [dotnet|305|305] [info]   DD_AZURE_APP_SERVICES=
10/30/20 02:43:14.415 PM [dotnet|305|305] [info]   APP_POOL_ID=
10/30/20 02:43:14.415 PM [dotnet|305|305] [info]   DOTNET_CLI_TELEMETRY_PROFILE=
10/30/20 02:43:14.452 PM [dotnet|306|306] [info] Profiler attached.
10/30/20 02:43:18.613 PM [dotnet|306|306] [info] *** JITCompilationStarted() replaced calls from StackExchange.Redis.RedisDatabase.KeyExpireAsync() to StackExchange.Redis.RedisBase.ExecuteAsync() 721420347 with calls to Datadog.Trace.ClrProfiler.Integrations.StackExchange.Redis.RedisBatch.ExecuteAsync() 721420489
10/30/20 02:43:18.645 PM [dotnet|306|306] [info] *** JITCompilationStarted() replaced calls from StackExchange.Redis.RedisDatabase.KeyDeleteAsync() to StackExchange.Redis.RedisBase.ExecuteAsync() 721420347 with calls to Datadog.Trace.ClrProfiler.Integrations.StackExchange.Redis.RedisBatch.ExecuteAsync() 721420489
10/30/20 02:43:18.774 PM [dotnet|305|305] [info] *** JITCompilationStarted() replaced calls from MongoDB.Driver.Core.WireProtocol.CommandWireProtocol`1.Execute() to MongoDB.Driver.Core.WireProtocol.IWireProtocol`1.Execute() 167772747 with calls to Datadog.Trace.ClrProfiler.Integrations.MongoDbIntegration.ExecuteGeneric() 167775891
10/30/20 02:43:18.806 PM [dotnet|306|306] [info] *** JITCompilationStarted() replaced calls from StackExchange.Redis.RedisDatabase.DebugObjectAsync() to StackExchange.Redis.RedisBase.ExecuteAsync() 721420354 with calls to Datadog.Trace.ClrProfiler.Integrations.StackExchange.Redis.RedisBatch.ExecuteAsync() 721420490
10/30/20 02:43:18.821 PM [dotnet|306|306] [info] *** JITCompilationStarted() replaced calls from StackExchange.Redis.RedisDatabase.ExecuteAsync() to StackExchange.Redis.RedisBase.ExecuteAsync() 721420387 with calls to Datadog.Trace.ClrProfiler.Integrations.StackExchange.Redis.RedisBatch.ExecuteAsync() 721420491
10/30/20 02:43:18.829 PM [dotnet|306|306] [info] *** JITCompilationStarted() replaced calls from StackExchange.Redis.RedisDatabase.GeoAddAsync() to StackExchange.Redis.RedisBase.ExecuteAsync() 721420347 with calls to Datadog.Trace.ClrProfiler.Integrations.StackExchange.Redis.RedisBatch.ExecuteAsync() 721420489
10/30/20 02:43:18.832 PM [dotnet|306|306] [info] *** JITCompilationStarted() replaced calls from StackExchange.Redis.RedisDatabase.GeoDistanceAsync() to StackExchange.Redis.RedisBase.ExecuteAsync() 721420358 with calls to Datadog.Trace.ClrProfiler.Integrations.StackExchange.Redis.RedisBatch.ExecuteAsync() 721420492
10/30/20 02:43:18.845 PM [dotnet|306|306] [info] *** JITCompilationStarted() replaced calls from StackExchange.Redis.RedisDatabase.GeoHashAsync() to StackExchange.Redis.RedisBase.ExecuteAsync() 721420362 with calls to Datadog.Trace.ClrProfiler.Integrations.StackExchange.Redis.RedisBatch.ExecuteAsync() 721420493
10/30/20 02:43:18.853 PM [dotnet|306|306] [info] *** JITCompilationStarted() replaced calls from StackExchange.Redis.RedisDatabase.GeoPositionAsync() to StackExchange.Redis.RedisBase.ExecuteAsync() 721420366 with calls to Datadog.Trace.ClrProfiler.Integrations.StackExchange.Redis.RedisBatch.ExecuteAsync() 721420494
10/30/20 02:43:18.862 PM [dotnet|306|306] [info] *** JITCompilationStarted() replaced calls from StackExchange.Redis.RedisDatabase.GeoRadiusAsync() to StackExchange.Redis.RedisBase.ExecuteAsync() 721420368 with calls to Datadog.Trace.ClrProfiler.Integrations.StackExchange.Redis.RedisBatch.ExecuteAsync() 721420495
10/30/20 02:43:18.865 PM [dotnet|306|306] [info] *** JITCompilationStarted() replaced calls from StackExchange.Redis.RedisDatabase.SortedSetRemoveAsync() to StackExchange.Redis.RedisBase.ExecuteAsync() 721420347 with calls to Datadog.Trace.ClrProfiler.Integrations.StackExchange.Redis.RedisBatch.ExecuteAsync() 721420489
10/30/20 02:43:18.867 PM [dotnet|306|306] [info] *** JITCompilationStarted() replaced calls from StackExchange.Redis.RedisDatabase.HashIncrementAsync() to StackExchange.Redis.RedisBase.ExecuteAsync() 721420374 with calls to Datadog.Trace.ClrProfiler.Integrations.StackExchange.Redis.RedisBatch.ExecuteAsync() 721420496
10/30/20 02:43:18.876 PM [dotnet|306|306] [info] *** JITCompilationStarted() replaced calls from StackExchange.Redis.RedisDatabase.HashDeleteAsync() to StackExchange.Redis.RedisBase.ExecuteAsync() 721420347 with calls to Datadog.Trace.ClrProfiler.Integrations.StackExchange.Redis.RedisBatch.ExecuteAsync() 721420489
10/30/20 02:43:18.877 PM [dotnet|306|306] [info] *** JITCompilationStarted() replaced calls from StackExchange.Redis.RedisDatabase.HashExistsAsync() to StackExchange.Redis.RedisBase.ExecuteAsync() 721420347 with calls to Datadog.Trace.ClrProfiler.Integrations.StackExchange.Redis.RedisBatch.ExecuteAsync() 721420489
10/30/20 02:43:18.877 PM [dotnet|306|306] [info] *** JITCompilationStarted() replaced calls from StackExchange.Redis.RedisDatabase.HashGetAllAsync() to StackExchange.Redis.RedisBase.ExecuteAsync() 721420371 with calls to Datadog.Trace.ClrProfiler.Integrations.StackExchange.Redis.RedisBatch.ExecuteAsync() 721420497
10/30/20 02:43:18.881 PM [dotnet|306|306] [info] *** JITCompilationStarted() replaced calls from StackExchange.Redis.RedisDatabase.HashKeysAsync() to StackExchange.Redis.RedisBase.ExecuteAsync() 721420372 with calls to Datadog.Trace.ClrProfiler.Integrations.StackExchange.Redis.RedisBatch.ExecuteAsync() 721420498
10/30/20 02:43:18.883 PM [dotnet|306|306] [info] *** JITCompilationStarted() replaced calls from StackExchange.Redis.RedisDatabase.HashLengthAsync() to StackExchange.Redis.RedisBase.ExecuteAsync() 721420356 with calls to Datadog.Trace.ClrProfiler.Integrations.StackExchange.Redis.RedisBatch.ExecuteAsync() 721420499
10/30/20 02:43:18.895 PM [dotnet|306|306] [info] *** JITCompilationStarted() replaced calls from StackExchange.Redis.RedisDatabase.HashSetAsync() to StackExchange.Redis.RedisBase.ExecuteAsync() 721420347 with calls to Datadog.Trace.ClrProfiler.Integrations.StackExchange.Redis.RedisBatch.ExecuteAsync() 721420489
10/30/20 02:43:18.907 PM [dotnet|306|306] [info] *** JITCompilationStarted() replaced calls from StackExchange.Redis.RedisDatabase.HashValuesAsync() to StackExchange.Redis.RedisBase.ExecuteAsync() 721420372 with calls to Datadog.Trace.ClrProfiler.Integrations.StackExchange.Redis.RedisBatch.ExecuteAsync() 721420498
10/30/20 02:43:18.908 PM [dotnet|306|306] [info] *** JITCompilationStarted() replaced calls from StackExchange.Redis.RedisDatabase.HyperLogLogAddAsync() to StackExchange.Redis.RedisBase.ExecuteAsync() 721420347 with calls to Datadog.Trace.ClrProfiler.Integrations.StackExchange.Redis.RedisBatch.ExecuteAsync() 721420489
```

@DataDog/apm-dotnet

### CODE REVIEW

File: src/Datadog.Trace.ClrProfiler.Native/Datadog.Trace.ClrProfiler.Native.vcxproj [hunk 1]
```diff
@@ -124,8 +146,17 @@
       <ConformanceMode>true</ConformanceMode>
       <SDLCheck>true</SDLCheck>
       <MultiProcessorCompilation>true</MultiProcessorCompilation>
-      <PreprocessorDefinitions>_TARGET_64BIT;_UNICODE;UNICODE;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+      <PreprocessorDefinitions>SPDLOG_FMT_EXTERNAL;_TARGET_64BIT;BIT64;_UNICODE;UNICODE;%(PreprocessorDefinitions)</PreprocessorDefinitions>
```
Comment (lucaspimentel) [inline src/Datadog.Trace.ClrProfiler.Native/Datadog.Trace.ClrProfiler.Native.vcxproj:149]: I see you added the missing `BIT64`. Nice catch. Can you add a note in the PR description that this PR also fixes the issue where the profiler log reports x86 even when the process is x64? (only affected `Release` builds). @kevingosse keeps bringing this up 😉

Comment (tonyredondo) [reply src/Datadog.Trace.ClrProfiler.Native/Datadog.Trace.ClrProfiler.Native.vcxproj:149]: copied and pasted 😁

**Review by lucaspimentel — APPROVED** [review]

**Review by zacharycmontoya — APPROVED** [review]

**Comment by zacharycmontoya** [comment]
Can you please add an entry to `LICENSE-3rdparty.csv` to indicate the usage of spdlog as well? Looks like it's MIT which is great

**Comment by tonyredondo** [comment]
> Can you please add an entry to `LICENSE-3rdparty.csv` to indicate the usage of spdlog as well? Looks like it's MIT which is great

Done in https://github.com/DataDog/dd-trace-dotnet/pull/1005/commits/5b88b76e716e80154847ec55465f9b6b59cda9c1

**Pull request merged**